### PR TITLE
Gold AJV and Internal endpoint hardening for review

### PR DESCRIPTION
**Pull request by afostr**

### CODE REVIEW

File: src/types/CompareCertReq.ts [hunk 1]
```diff
@@ -28,5 +29,11 @@ export const deserializeCompareCertReq = (stream: VectorBufferStream): CompareCe
     throw new Error(`Unsupported CompareCertReqSerializable version ${version}`)
   }
   const obj: CompareCertReqSerializable = Utils.safeJsonParse(stream.readString())
+
+  const errors = verifyPayload('CompareCertReq', obj)
```
Comment (ahmxdiqbal) [inline src/types/CompareCertReq.ts]: Why isnt `AJVSchemaEnum...` being used here like it is in ApoptosisProposalReq.ts?

File: src/types/CompareCertResp.ts [hunk 2]
```diff
@@ -29,5 +30,10 @@ export const deserializeCompareCertResp = (stream: VectorBufferStream): CompareC
 
   const obj: CompareCertRespSerializable = Utils.safeJsonParse(stream.readString())
 
+  const errors = verifyPayload('CompareCertResp', obj)
```
Comment (ahmxdiqbal) [inline src/types/CompareCertResp.ts]: Why isnt `AJVSchemaEnum...` being used here like it is in ApoptosisProposalReq.ts?

**Review by afostr — COMMENTED** [review]
Got to the end, looks great to me

File: test/unit/src/types/SendCachedAppDataReq.test.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+const cSendCachedAppDataReqVersion = 1
+
+describe('SendCachedAppDataReq Serialization and Deserialization', () => {
+    beforeAll(() => {
+        initAjvSchemas()
+    })
+
+    describe('serializeSendCachedAppDataReq Serialization', () => {
+        test('should serialize with root true', () => {
+            const obj: SendCachedAppDataReq = {
```
Comment (afostr) [inline test/unit/src/types/SendCachedAppDataReq.test.ts]: note, all unit tests pass except for this one. starts with a compile error on this line.

File: src/types/CompareCertReq.ts (hunk 1 above)
Comment (arhamj) [reply src/types/CompareCertReq.ts]: Fixed.

File: src/types/CompareCertResp.ts (hunk 2 above)
Comment (arhamj) [reply src/types/CompareCertResp.ts]: Fixed

File: test/unit/src/types/SendCachedAppDataReq.test.ts (hunk 3 above)
Comment (arhamj) [reply test/unit/src/types/SendCachedAppDataReq.test.ts]: Fixed

**Review by mssabr01 — APPROVED** [review]

**Pull request merged**